File: Cap 06/6.2 - Matrizes de Uma Dimensao/Media Geral V3/medverv3.ts
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 6 - Topico 6.2
 * 
 * Programa ..: MEDIA GERAL - VERSAO 3
 */

import * as readlineSync from 'readline-sync';

let MD: number[] = [];
let SOMA: number = 0;

console.log("MEDIA GERAL - V3\n");

for (let I: number = 0; I <= 7; I++) {
  MD[I] = parseFloat(readlineSync.question("Entre a nota " + (I + 1).toString().padStart(2, " ") + ": "));
  SOMA = SOMA + MD[I];
}

let MEDIA: number = SOMA / 8;

for (let I: number = 0; I <= 7; I++) {
  console.log("A[" + I.toString().padStart(2, "0") + "] = " + MD[I].toFixed(1));
}

console.log("Resultado da media = " + MEDIA.toFixed(1));

File: Cap 06/6.2 - Matrizes de Uma Dimensao/Media Geral V3/medverv3.vb
' Livro .....: Algorimos
'              Logica para Desenvolvimento de Programacao de 
'              Computadores
' Autores ...: Jose Augusto N. G. Manzano
'              Jayr Figueiredo de Oliveira
' Editora ...: Erica
' 
' Capitulo ..: 6 - Topico 6.2
' 
' Programa ..: MEDIA GERAL - VERSAO 3

Module MediaGeral

    Sub Main()

        Dim I As Integer
        Dim MD(0 To 7) As Single
        Dim SOMA As Single, MEDIA As Single

        Console.WriteLine("MEDIA GERAL - V3")
        Console.WriteLine()

        SOMA = 0
        For I = 0 To 7
            Console.Write("Entre a nota {0}: ", I + 1)
            MD(I) = Console.ReadLine()
            SOMA = SOMA + MD(I)
        Next I

        MEDIA = SOMA / 8

        For I = 0 To 7
            Console.WriteLine("A[{0}] = {1:###.#}", Right(" " & I + 1, 2), MD(I))
        Next I

        Console.WriteLine("Resultado da media = {0:###.#}", MEDIA)

    End Sub

End Module
